p2p/host.go

```diff
@@ -70,24 +64,29 @@ func NewHost(self *Peer, key libp2p_crypto.PrivKey) (Host, error) {
 		return nil, errors.Wrapf(err,
 			"cannot create listen multiaddr from port %#v", self.Port)
 	}
+
 	ctx := context.Background()
 	p2pHost, err := libp2p.New(ctx,
-		libp2p.ListenAddrs(listenAddr), libp2p.Identity(key),
+		libp2p.ListenAddrs(listenAddr),
+		libp2p.Identity(key),
+		libp2p.EnableNATService(),
+		libp2p.ForceReachabilityPublic(),
 	)
 	if err != nil {
 		return nil, errors.Wrapf(err, "cannot initialize libp2p host")
 	}
-	traceFile := os.Getenv("P2P_TRACEFILE")
 
-	const MaxSize = 2_145_728
 	options := []libp2p_pubsub.Option{
-		libp2p_pubsub.WithPeerOutboundQueueSize(64),
-		libp2p_pubsub.WithMaxMessageSize(MaxSize),
-	}
-	if len(traceFile) > 0 {
-		tracer, _ := libp2p_pubsub.NewJSONTracer(traceFile)
-		options = append(options, libp2p_pubsub.WithEventTracer(tracer))
+		// WithValidateQueueSize sets the buffer of validate queue. Defaults to 32. When queue is full, validation is throttled and new messages are dropped.
+		libp2p_pubsub.WithValidateQueueSize(64),
+		// WithPeerOutboundQueueSize is an option to set the buffer size for outbound messages to a peer. We start dropping messages to a peer if the outbound queue if full.
+		libp2p_pubsub.WithPeerOutboundQueueSize(32),
```

Comment:
Same for this actually, this hotspot node (like leader) may experience msg drops.

Reply:
this is outbound queue size, even leader doesn't need to have a large queue for outbound unless leader has to send a lot of messages, IMHO. Leader will receive a lot of messages, but not outbound though.  We can double it for now.  But I'll dig deeper into the metrics of p2p to be able to find some more data, so that we can tune those number later though.